scripts/globals/magic.lua

```diff
@@ -340,19 +341,27 @@ function applyResistance(player,spell,target,diff,skill,bonus)
     if player:hasStatusEffect(EFFECT_ALTRUISM) and spell:getSpellGroup() == SPELLGROUP_WHITE then
         magicacc = magicacc + player:getStatusEffect(EFFECT_ALTRUISM):getPower();
     end
+	
     if player:hasStatusEffect(EFFECT_FOCALIZATION) and spell:getSpellGroup() == SPELLGROUP_BLACK then
         magicacc = magicacc + player:getStatusEffect(EFFECT_FOCALIZATION):getPower();
     end
     --difference in int/mnd
+	
     if (diff > 10) then
         magicacc = magicacc + 10 + (diff - 10)/2;
     else
         magicacc = magicacc + diff;
     end
-    --add acc for ele/dark seal
+	
+    --Add acc for dark seal
     if (player:getStatusEffect(EFFECT_DARK_SEAL) ~= nil and skill == DARK_MAGIC_SKILL) then
         magicaccbonus = magicaccbonus + 256;
     end
+	
+    --Add acc for klimaform
+    if (player:getStatusEffect(EFFECT_KLIMAFORM) and (castersWeather == singleWeatherStrong[element] or castersWeather == doubleWeatherStrong[element])) then

```

Comment:
oh< I didn't even realized he had get instead of has.

Its hasStatusEffect(0 when you just want to knwo if they have the effect or not. getStatusEffect() when you need to get info about the effect like power/subpower etc.